driver-sync/src/main/com/mongodb/client/internal/MongoDatabaseImpl.java

```diff
@@ -199,8 +194,7 @@ private <TResult> TResult executeCommand(@Nullable final ClientSession clientSes
         if (clientSession != null && clientSession.hasActiveTransaction() && !readPreference.equals(ReadPreference.primary())) {
             throw new MongoClientException("Read preference in a transaction must be primary");
         }
-        return executor.execute(new CommandReadOperation<TResult>(getName(), toBsonDocument(command), codecRegistry.get(resultClass)),
-                readPreference, readConcern, clientSession);
+        return executor.execute(operations.runCommand(command, resultClass), readPreference, readConcern, clientSession);
```

Comment:
`executor.execute(operations.runCommand...` is really confusing as `runCommand` looks like a synonym to `execute`. Can we rename the `runCommand` method to `read` (it creates `CommandReadOperation`) or something like that?

Reply:
For consistency I went with `commandRead`